CMakeLists.txt

```
@@ -0,0 +1,66 @@
+cmake_minimum_required(VERSION 3.5)
+project(noether_ros)
+
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+# find dependencies
+find_package(ament_cmake REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(geometry_msgs REQUIRED)
+find_package(tf2_eigen REQUIRED)
+find_package(noether_tpp REQUIRED)
+find_package(yaml-cpp REQUIRED)
+
+# Generate the messages
+find_package(rosidl_default_generators REQUIRED)
+rosidl_generate_interfaces(
+  ${PROJECT_NAME}
+  "msg/ToolPath.msg"
+  "msg/ToolPaths.msg"
+  "srv/PlanToolPath.srv"
+  DEPENDENCIES
+  builtin_interfaces
+  geometry_msgs
+)
+rosidl_get_typesupport_target(interfaces_target "${PROJECT_NAME}" "rosidl_typesupport_cpp")
+
+# Library
+add_library(${PROJECT_NAME}_conversions SHARED src/conversions.cpp)
+target_include_directories(
+  ${PROJECT_NAME}_conversions PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+                                     "$<INSTALL_INTERFACE:include>"
+)
```

This may have just been what cmake-format did, but it seems like this would look cleaner and be a bit easier to read:
```suggestion
target_include_directories(${PROJECT_NAME}_conversions PUBLIC
  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
  "$<INSTALL_INTERFACE:include>")
```

---

Agreed, but this is the result of `cmake-format` and I don't know how to force it to do this

---

The creators of cmake-format seem to agree that it is hard to force cmake-format to do this, as they have an extensive documentation page showing different cases that don't really work well with their simple rule options:

https://cmake-tools-staging.readthedocs.io/en/stable/format-case_studies.html

Oh, well.